tools/src/main/java/org/apache/kafka/tools/ShareConsumerPerformance.java

```diff
@@ -0,0 +1,327 @@
+                .withOptionalArg()
+                .describedAs("milliseconds")
+                .ofType(Long.class)
+                .defaultsTo(10_000L);
+            numMessagesOpt = parser.accepts("messages", "REQUIRED: The number of messages to send or consume")
+                .withRequiredArg()
+                .describedAs("count")
+                .ofType(Long.class);
+            reportingIntervalOpt = parser.accepts("reporting-interval", "Interval in milliseconds at which to print progress info.")
+                .withRequiredArg()
+                .withValuesConvertedBy(regex("^\\d+$"))
+                .describedAs("interval_ms")
+                .ofType(Long.class)
+                .defaultsTo(5_000L);
+            dateFormatOpt = parser.accepts("date-format", "The date format to use for formatting the time field. " +
+                    "See java.text.SimpleDateFormat for options.")
+                .withRequiredArg()
+                .describedAs("date format")
+                .ofType(String.class)
+                .defaultsTo("yyyy-MM-dd HH:mm:ss:SSS");
+            hideHeaderOpt = parser.accepts("hide-header", "If set, skips printing the header for the stats");
+            try {
+                options = parser.parse(args);
+            } catch (OptionException e) {
+                CommandLineUtils.printUsageAndExit(parser, e.getMessage());
+                return;
+            }
+            if (options != null) {
+                if (options.has(numThreadsOpt) || options.has(numFetchersOpt))
+                    System.out.println("WARNING: option [threads] and [num-fetch-threads] have been deprecated and will be ignored by the test");
+                CommandLineUtils.maybePrintHelpOrVersion(this, "This tool is used to verify the share consumer performance.");
+                CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, numMessagesOpt);
+            }
+        }
+
+        public boolean printMetrics() {
+            return options.has(printMetricsOpt);
+        }
+
+        public String brokerHostsAndPorts() {
+            return options.valueOf(options.has(bootstrapServerOpt) ? bootstrapServerOpt : brokerListOpt);
+        }
+
+        public Properties props() throws IOException {
+            Properties props = (options.has(consumerConfigOpt))
+                ? Utils.loadProps(options.valueOf(consumerConfigOpt))
+                : new Properties();
+            props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerHostsAndPorts());
+            props.put(ConsumerConfig.GROUP_ID_CONFIG, options.valueOf(groupIdOpt));
+            props.put(ConsumerConfig.RECEIVE_BUFFER_CONFIG, options.valueOf(socketBufferSizeOpt).toString());
+            props.put(ConsumerConfig.MAX_PARTITION_FETCH_BYTES_CONFIG, options.valueOf(fetchSizeOpt).toString());
+            props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
```

Comment:
Doesn't apply to share consumer.

Reply:
depends on the discussion of comment https://github.com/confluentinc/kafka/pull/1297#discussion_r1642376578

Reply:
I think you need to remove this option and the command line option which affects it. Then I'm ready to approve.